libraries/AP_HAL_ChibiOS/CANClock.cpp

```diff
@@ -0,0 +1,483 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ */
```

Comment:
for the files that are derived from uavcan we need to have a copy of the MIT license, as required by that license. This could be done by putting a copy of the license (it is quite short) in each file, or by putting these files in a subdirectory and having a README.md in that directory explaining where the code came from and a copy of the license 
Note that we can mark them as GPLv3, but we also need a copy of the MIT license in there, with a statement saying "portions of this code are based on code by Pavel Kirienko under the MIT license (with a copy of the license)"
This is required because of the following lines in the MIT license:

> The above copyright notice and this permission notice shall be included in all
> copies or substantial portions of the Software.

Reply:
alternatively, you could ask permission from Pavel to use it directly under GPLv3 without reference to the MIT license, but that would be a bit unusual, and there is not good reason to do it